### PR TITLE
sqlite3: various improvements

### PR DESCRIPTION
**Pull request by micmac1**
Maintainer: me
Compile tested: ath79
Run tested: not yet. Will have access to trunk snapshot running on router this coming weekend. Will test then and ping back.

Description:
Hi all,

Some commits I came up with which I think would be of general benefit. If you feel strongly about any of this please let me know.

Kind regards,
Seb

### CODE REVIEW

File: libs/sqlite3/Makefile [hunk 1]
```diff
@@ -75,31 +92,46 @@ $(call Package/sqlite3/Default/description)
...
 
-ifneq ($(CONFIG_SQLITE_RTREE),y)
-TARGET_CFLAGS += -USQLITE_ENABLE_RTREE
-endif
+TARGET_CFLAGS += \
+	-DHAVE_ISNAN \
+	-DHAVE_MALLOC_USABLE_SIZE \
+	-DSQLITE_ENABLE_UNLOCK_NOTIFY \
+	-DSQLITE_LIKE_DOESNT_MATCH_BLOBS \
+	-DSQLITE_MAX_SCHEMA_RETRY=25 \
```
Comment (diizzyy) [inline libs/sqlite3/Makefile]: Why should we change default value?

**Comment by diizzyy** [comment]
Drop JSON and ```-DSQLITE_LIKE_DOESNT_MATCH_BLOBS`` as defaults unless there's a very good reason to keep those.

**Comment by micmac1** [comment]
JSON is default. DSQLITE_LIKE_DOESNT_MATCH_BLOBS is an optimization.

Feel free to consult https://www.sqlite.org/compile.html.

**Comment by diizzyy** [comment]
Yes, but do we need JSON or does it add anything noticably to the binary size?
https://svnweb.freebsd.org/ports/head/databases/sqlite3/Makefile?revision=486622&view=markup#l136
At least FreeBSD disables it by default

While MATCH_BLOBS is an optimization do we really know that all users of sqlite will work with it enabled?

**Comment by micmac1** [comment]
Removing json support were not gonna do.

As far as the optimization of LIKE goes, seeing that Debian uses it IMHO it's safe to say there's no danger here.

Happy holidays.

**Comment by micmac1** [comment]
*we're

**Comment by micmac1** [comment]
Hello,

I removed the third commit that tried to introduce some optimizations. It's not important.

To summarize, these commits add new menuconfig options and clean up the Makefile. For the default library build nothing changes (the feature set before and after this pull request ist the same.

For the cli tool the command-line editing interface is changed from readline to libedit, which is upstream's first choice and is also a tad smaller than readline.

Run-tested on ath79, running asterisk 13 against libsqlite and also running the cli tool against the db.

Please merge.

Kind regards,
Seb

**Pull request merged**